File: else/bronze/2/2920/lua/2920.lua
local isAscending = true
local isDescending = true

local prev = 0
for i = 1, 8 do
    local num
    num = io.read("*n");

    if prev ~= 0 and prev + 1 ~= num then
        isAscending = false
    end
    if prev ~= 0 and prev - 1 ~= num then
        isDescending = false
    end
    prev = num
end

if isAscending then
    print("ascending")
elseif isDescending then
    print("descending")
else
    print("mixed")
end

File: else/bronze/2/2920/c/2920.c
#include <stdio.h>

int main() {
    int isAscending = 1;
    int isDescending = 1;

    int prev = 0;
    for(int i = 0; i < 8; i++) {
        int num;
        scanf("%d", &num);

        if(prev != 0 && prev + 1 != num) {
            isAscending = 0;
        }
        if(prev != 0 && prev - 1 != num) {
            isDescending = 0;
        }
        prev = num;
    }

    if(isAscending) {
        printf("%s", "ascending");
    }
    else if(isDescending) {
        printf("%s", "descending");
    }
    else {
        printf("%s", "mixed");
    }

    return 0;
}